cinnabar/tests/test_stats.py

```diff
@@ -145,3 +146,71 @@ def test_correlation_positive(fe_map):
         assert (
             0.5 < bss["mle"] < 0.9
         ), f"Correlation must be positive for this data. {stat} is {bss['mle']}"
+
+
+@pytest.fixture(scope='module')
+def example_data(fe_map):
+    """
+    Returns data w/ error bars from `cinnabar/data/example.csv`
+    """
+    nodes = fe_map.graph.nodes
+
+    x_data = np.asarray([n[1]["exp_DG"] for n in nodes(data=True)])
+    y_data = np.asarray([n[1]["calc_DG"] for n in nodes(data=True)])
+    xerr = np.asarray([n[1]["exp_dDG"] for n in nodes(data=True)])
+    yerr = np.asarray([n[1]["calc_dDG"] for n in nodes(data=True)])
+    
+    return x_data, y_data, xerr, yerr
+
+
+def test_confidence_intervals_defaults(example_data):
+    """
+    Test that boostrap confidence intervals contains
+    the 'mle' value when using defaults.
+    """
+    error_message =  "The stat must lie within the bootstrapped 95% CI"
+    x_data, y_data, xerr, yerr = example_data
+    bss = bootstrap_statistic(x_data, y_data, xerr, yerr, statistic="RMSE")
+    assert bss['low'] < bss['mle'] < bss['high'], error_message
+
+
+@pytest.mark.parametrize('stat,true_uncert,pred_uncert,estimate', [
+    ['RMSE', False, False, 'mle'],
+    ['MUE', False, False, 'mle'],
+    ['RMSE', True, False, 'mean'],
+    ['RMSE', False, True, 'mean'],
+    ['RMSE', True, True, 'mean'],
+])
+def test_confidence_intervals(example_data, stat, true_uncert,
+                              pred_uncert, estimate):
+    """
+    Test that the bootstrapped confidence intervals contain the
+    corresponding statistics.
+    Uses the example data in `cinnabar/data/example.csv`
+    """
+    error_message =  "The stat must lie within the bootstrapped 95% CI"
+    x_data, y_data, xerr, yerr = example_data
+    bss = bootstrap_statistic(x_data, y_data, xerr, yerr, statistic=stat,
+                              include_true_uncertainty=true_uncert,
+                              include_pred_uncertainty=pred_uncert)
+    assert bss['low'] < bss[estimate] < bss['high'], error_message
+
+
+def test_confidence_interval_edge_case():
+    """
+    Test that the bootstrapped confidence interval
+    for RMSE contains the sample estimate.
+    Uses the data from https://github.com/OpenFreeEnergy/cinnabar/issues/73
+    """
+
+    # Data from Cinnabar issue #73
+    x_data = [-0.101, 0.351, 0.117, 0.623, 5.172, 5.209, -1.727, -1.387, -1.534, 1.082]
+    y_data = [-0.174, 0.42, 0.262, 0.626, 5.064, 4.783, -1.58, -1.712, -1.699, 0.822]
+    xerr = [0.443, 0.652, 0.57, 0.245, 1.112, 1.049, 1.23, 1.435, 1.521, 0.505]
+    yerr = [0.442, 0.714, 0.619, 0.224, 1.401, 1.107, 1.178, 1.252, 1.265, 0.472]
+
+    # RMSE (default mode)
+    bss = bootstrap_statistic(x_data, y_data, xerr, yerr, statistic="RMSE",
+                              include_true_uncertainty=True,
+                              include_pred_uncertainty=True)
+    assert (bss['low'] > bss['mle']) or (bss['mle'] > bss['high']), error_message
```

Comment:
Was reviewing @IAlibay 's changes and just realized that `error_message` isn't defined here. I'll open a PR to fix this.

Reply:
Here's the PR: https://github.com/OpenFreeEnergy/cinnabar/pull/76
Note that I also found another issue with the edge case test and fixed it in the above PR